pkg/v1/mutate/index.go

```diff
@@ -56,6 +57,10 @@ func computeDescriptor(ia IndexAddendum) (*v1.Descriptor, error) {
 type index struct {
 	base v1.ImageIndex
 	adds []IndexAddendum
+	// replaces are applied before adds
+	replaces []v1.Descriptor
+	// removes are removed after replaces and before adds
+	removes []match.Matcher
```

Comment:
I don't think this needs to be a list (yet?). In theory, we may be able to optimize some things in the future to do introspection and directly modify these fields instead of returning an entirely new `mutate.index` each time, but right now this can only be a single matcher, I think.

Reply:
It probably can, but it parallels adds, for a clean API, and I made it similarly variadic. Let's just leave it.

Reply:
Eh... I'm not sure.

I do like that this mirrors AddManifests, but I think making this variadic is going to be a source of confusion. There's more than one way to compose matchers, and this interface forces the operator to only ever be "or".

You can achieve the same thing now with multiple calls to `RemoveManifests`, which is definitely less efficient, but you can also just implement a `match.Matcher` that matches multiple predicates.

Having `RemoveManifests` accept multiple matchers encodes some matcher composition within mutate, and I'd rather that live in `match` if we need it.

Of course, you know that I will advocate for adding `match.Or` and `match.And` to solve this, but we're not there yet :P

Reply:
OK done. And rebased.